### PR TITLE
feat(android): add openFile to utils

### LINKED ISSUE #5661: [Feature] Add openFile method for android
### Did you verify this is a real problem by searching the [NativeScript Forum](http://forum.nativescript.org) and the [other open issues in this repo](https://github.com/NativeScript/nativescript/issues)?
Yes

### Tell us about the problem
Currently under iOS we can easily open a file by using `utils.ios.openFile()` method. Under android there is no similar method. There should be a similar method for android enabling the user to open local files. Code similar to this can be used in order to achieve it:
https://github.com/PeterStaev/nativescript-telerik-reporting/blob/b57db9e79f4c4c7ef71fdb070cebd3063e106995/demo/app/main-page.ts#L85-L91
The only catch is that the file must be present on the external storage in order other apps to have access to it. 

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 3.4
- Cross-platform modules: 3.4
- Runtime(s): 3.4

### PR DESCRIPTION
**Pull request by t3hmrman**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [ ] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?
No `openFile` functionality in android utils

## What is the new behavior?
Adds `openFile` to `utils` android module.

Fixes/Implements/Closes #5661.

### CODE REVIEW

**Comment by t3hmrman** [comment]
I'm not sure how to unit test this change, as `openFile` will open another activity -- I don't see a similar test for ios.

The test for android works (and opens a file chooser) but in doing so it puts the app into the background which makes subsequent tests fail...

File: tests/app/utils/utils-tests.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import * as TKUnit from "../TKUnit";
 import * as utils from "tns-core-modules/utils/utils";
-import { isIOS } from "tns-core-modules/platform";
+import * as fs from "tns-core-modules/file-system";
+import { isIOS, isAndroid } from "tns-core-modules/platform";
+import { getNativeApplication } from "tns-core-modules/application";
+import * as frame from "tns-core-modules/ui/frame";
+import * as application from "tns-core-modules/application";;
```
Comment (t3hmrman) [inline tests/app/utils/utils-tests.ts]: Just noticed these imports -- they were added while I was trying to figure out a way to finish out the test after the file chooser intent opened... They can mostly be removed safely.

**Comment by vakrilov** [comment]
@cla-bot check

**Comment by t3hmrman** [comment]
Hey thanks for taking a look!

That looks good to me, please feel free to commit to the PR.

Thanks for all the hard work on Nativescript!

**Comment by ADjenkov** [comment]
test

**Comment by t3hmrman** [comment]
Hey what about the fact that the test causes the app to be unresponsive (since it opens the chooser)?

File: tests/app/utils/utils-tests.ts (hunk 1 above)
Comment (t3hmrman) [reply tests/app/utils/utils-tests.ts]: Resolving since this was resolved by @ADjenkov

**Comment by ADjenkov** [comment]
> Hey what about the fact that the test causes the app to be unresponsive (since it opens the chooser)?
> […](#)
> On Mar 1, 2019, at 23:22, Alexander Djenkov ***@***.***> wrote: test — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

We'll later extend the test, with `nativescript-dev-appium`, to click on the appropriate icon from the chooser.

**Pull request merged**